### PR TITLE
Install script for MacOS/M1 and README update

### PR DESCRIPTION
**Pull request by J0HNB0Y**
Script to install the HPE GL provider on MacOS for Intel and Apple Silicon devices.

Installs the provider in `~/.terraform.d/plugins` in line with the new Terraform file system layout (found [here](https://www.terraform.io/upgrade-guides/0-13.html#new-filesystem-layout-for-local-copies-of-providers))

### CODE REVIEW

File: tools/install-hpegl-provider-macos-m1.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+os="darwin"
+arch="arm64"
```
Comment (zachovich) [inline tools/install-hpegl-provider-macos-m1.sh:6]: can we detect the underlying arch and set it instead of having two separate scripts?

Comment (J0HNB0Y) [reply tools/install-hpegl-provider-macos-m1.sh]: Added architecture detection

File: tools/install-hpegl-provider-macos.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -e
+
+os="darwin"
+
+if [[ `uname -m` == 'x86_64']]
```
Comment (zachovich) [inline tools/install-hpegl-provider-macos.sh:7]: isn't a space required be there before closing ]]

Comment (J0HNB0Y) [reply tools/install-hpegl-provider-macos.sh]: Yes :)

**Review by zachovich — APPROVED** [review]

**Pull request merged**